src/transformers/configuration_utils.py

```diff
@@ -46,6 +46,35 @@
 _re_configuration_file = re.compile(r"config\.(.*)\.json")
 
 
+GENERATION_DEFAULTS = {
```

Comment:
Not a global mutable! 🙈  

Even though it's treated properly in the pretrained config - I can completely imagine this being modified and causing head-scratching bugs around the place. Let's wrap this up in a function

```
def get_generation_defaults():
    return {
        "max_length": 20, 
        ...
    }
```